src/request-pipeline/header-transforms/index.ts

```diff
@@ -54,6 +54,13 @@ export function transformHeadersCaseToRaw (headers: OutgoingHttpHeaders, rawHead
         }
     }
 
+    // NOTE: We doesn't send a cross-domain client request as cross-domain.
+    // Therefore, the origin header can be absent and we cannot decide its case. GH-2382
+    if (headers.hasOwnProperty('origin') && processedHeaders.hasOwnProperty('Referer')) {
```

Comment:
Since it's an edge case, let's move it to the separate function and put the comment into it.
```js
...
calculateOriginHeaderCase (headers, processedHeaders, headersNames);
...
```